src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -893,10 +900,57 @@ private void FlushPendingOutput()
-        public abstract PSCredential PromptForCredential(string caption, string message,
-            string userName, string targetName, PSCredentialTypes allowedCredentialTypes,
-            PSCredentialUIOptions options
-        );
+        /// <seealso cref="System.Management.Automation.Host.PSHostUserInterface.PromptForCredential(string, string, string, bool, string, System.Management.Automation.PSCredentialTypes, System.Management.Automation.PSCredentialUIOptions)"/>
+        public abstract PSCredential PromptForCredential(
+            string caption,
+            string message,
+            string userName,
+            string targetName,
+            PSCredentialTypes allowedCredentialTypes,
+            PSCredentialUIOptions options);
+
+        /// <summary>
+        /// Prompt for credential.
+        /// </summary>
+        /// <param name="caption">
+        /// Caption for the message.
+        /// </param>
+        /// <param name="message">
+        /// Text description for the credential to be prompt.
+        /// </param>
+        /// <param name="userName">
+        /// Name of the user whose credential is to be prompted for. If set to null or empty
+        /// string, the function will prompt for user name first.
+        /// </param>
+        /// <param name="confirmPassword">
+        /// Prompts user to re-enter the password for confirmation.
+        /// </param>
+        /// <param name="targetName">
+        /// Name of the target for which the credential is being collected.
+        /// </param>
+        /// <param name="allowedCredentialTypes">
+        /// Types of credential can be supplied by the user.
+        /// </param>
+        /// <param name="options">
+        /// Options that control the credential gathering UI behavior.
+        /// </param>
+        /// <returns>
+        /// User input credential.
+        /// </returns>
+        /// <seealso cref="System.Management.Automation.Host.PSHostUserInterface.ReadLine"/>
+        /// <seealso cref="System.Management.Automation.Host.PSHostUserInterface.ReadLineAsSecureString"/>
+        /// <seealso cref="System.Management.Automation.Host.PSHostUserInterface.Prompt"/>
+        /// <seealso cref="System.Management.Automation.Host.PSHostUserInterface.PromptForChoice"/>
+        /// <seealso cref="System.Management.Automation.Host.PSHostUserInterface.PromptForCredential(string, string, string, string)"/>
+        /// <seealso cref="System.Management.Automation.Host.PSHostUserInterface.PromptForCredential(string, string, string, string, System.Management.Automation.PSCredentialTypes, System.Management.Automation.PSCredentialUIOptions)"/>
+        public abstract PSCredential PromptForCredential(
+            string caption,
+            string message,
+            string userName,
+            bool confirmPassword,
+            string targetName,
+            PSCredentialTypes allowedCredentialTypes,
+            PSCredentialUIOptions options);
```

Comment:
@TylerLeonhardt this may be why the PowerShell daily build has been causing CI issues.

There's a new abstract method in the host which has broken me:

```
An error has occurred that was not properly handled. Additional information is shown below. The PowerShell process will exit.
Unhandled exception. System.TypeLoadException: Method 'PromptForCredential' in type 'Microsoft.PowerShell.EditorServices.Services.PowerShell.Host.EditorServicesConsolePSHostUserInterface' from assembly 'Microsoft.PowerShell.EditorServices, Version=2.2.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at Microsoft.PowerShell.EditorServices.Services.PowerShell.Host.EditorServicesConsolePSHost..ctor(ILoggerFactory loggerFactory, String name, Version version, PSHost internalHost, ConsoleReadLine readline)
   at Microsoft.PowerShell.EditorServices.Services.PowerShell.PowerShellExecutionService.Initialize() in C:\Users\Robert Holt\Documents\Dev\Microsoft\PowerShellEditorServices\src\PowerShellEditorServices\Services\PowerShell\PowerShellExecutionService.cs:line 286
   at Microsoft.PowerShell.EditorServices.Services.PowerShell.PowerShellExecutionService.RunTopLevelConsumerLoop() in C:\Users\Robert Holt\Documents\Dev\Microsoft\PowerShellEditorServices\src\PowerShellEditorServices\Services\PowerShell\PowerShellExecutionService.cs:line 244
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

I'll open an issue to track fixing this

Reply:
Can we add new CI for ESS in the repo?

Reply:
@rjmholt , @iSazonov  Anything I could do to help ?

Reply:
@Shriram0908 Thanks! It is a question for MSFT team - they could add optional CIs for dependent tools so we could see possible regressions early.